### PR TITLE
Allow `clippy::type_complexity` in more places.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
# Objective

- See fewer warnings when running `cargo clippy` locally.

## Solution

- allow `clippy::type_complexity` in more places, which also signals to users they should do the same.

### CODE REVIEW

**Comment by IceSentry** [comment]
When running clippy in CI we use pre configured lints because we don't want to pollute the repo with a bunch of annotations like that.

Personally, I added this config to my rust analyzer to use the same lints as bevy

```json
   "rust-analyzer.check.command": "clippy",
   "rust-analyzer.check.extraArgs": [
      "--all-features",
      "--",
      "-Aclippy::type_complexity",
      "-Wclippy::doc_markdown",
      "-Wclippy::redundant_else",
      "-Wclippy::match_same_arms",
      "-Wclippy::semicolon_if_nothing_returned",
      "-Wclippy::explicit_iter_loop",
      "-Wclippy::map_flatten",
      "-Dwarnings"
  ],
```

**Comment by waywardmonkeys** [comment]
I thought about that, but there were already about 37 of these in the code (every crate basically except the new asset crate).

**Comment by waywardmonkeys** [comment]
I'll update this. To avoid a conflict, we'll need #9794 to land before this.

**Comment by waywardmonkeys** [comment]
I removed `-A clippy::type_complexity` from CI as discussed above and elsewhere. If examples change in such a way that they start to trigger, then they would need to be updated since this PR doesn't edit every single example, only those that currently trigger the lint.

File: examples/3d/anti_aliasing.rs [hunk 1]
```diff
@@ -1,5 +1,9 @@
 //! This example compares MSAA (Multi-Sample Anti-aliasing), FXAA (Fast Approximate Anti-aliasing), and TAA (Temporal Anti-aliasing).
 
+// This lint usually gives bad advice in the context of Bevy -- hiding complex queries behind
+// type alias tends to obfuscate code while offering no improvement in code cleanliness.
```
Comment (JoJoJet) [inline examples/3d/anti_aliasing.rs:4]:
```suggestion
// This lint usually gives bad advice in the context of Bevy -- hiding complex queries behind
// type aliases tends to obfuscate code while offering no improvement in code cleanliness.
```

I had a grammatical error in my original suggestion.

Comment (waywardmonkeys) [reply examples/3d/anti_aliasing.rs]: I should've noticed that. Fixed and force pushed.

**Review by JoJoJet — APPROVED** [review]

**Review by james7132 — APPROVED** [review]

**Comment by waywardmonkeys** [comment]
Sure thing. I am almost done with dinner and then will do it

**Comment by waywardmonkeys** [comment]
@alice-i-cecile Done! Hopefully everything is correct with the latest push.

**Pull request merged**